[Feat/application_view]application_view 작성

Kang9805 opened the pull request:
## ✅ PR 요약
- 지원서 작성/조회/취소 기능 구현 및 Serializer 수정, URL 구조 개편, test code 작성

## 📄 상세 내용
- [ ] Applicant 전용 View 기능 구현, test code 구현
- [ ] application serializer, admin serializer 수정
- [ ] Application urls 매핑
- [ ] 불필요 파일 및 dead code 정리
- [ ] applied_at 제거 후 created_at으로 통일

## 🧪 PR Checklist
- [ ] 커밋 메시지 컨벤션에 맞게 작성했습니다.
- [ ] 변경 사항에 대한 테스트를 했습니다.(버그 수정/기능에 대한 테스트).

Kang9805 committed:
✨ feat : application 파일 구조 정리

application 파일 구조 정리

Kang9805 committed:
♻️ refactor : application_serializer, admin_application_serializer applied_at 매핑 오류 수정

- applied_at 필드 매핑 오류 수정
- Applicant/Admin 시리얼라이저 필드 일관성 정리

Kang9805 committed:
✨ feat : 지원 취소용 application serializer 추가

취소 요청 성공 시 uuid, status만 반환하도록 전용 시리얼라이저 분리 구현

Kang9805 committed:
♻️ refactor : application 시리얼라이저 구조 개편 및 공통 필드 정리

- cancel 전용 Serializer 파일 분리 및 적용
- 타입 명시 및 Swagger 대응 코드 변경

Kang9805 committed:
✨ feat : 지원자(applicant)관련 view 로직 및 swagger 문서화

지원서 제출, 목록 조회, 상세 조회, 취소 view 생성

Kang9805 committed:
♻️ refactor : application 도메인 url 구조 정리 및 applicant view 연결

Kang9805 committed:
💡 chore : 사용하지 않는 service 레이어 제거, views.py 변경, models/init 파일 정리

사용하지 않는 service 레이어 제거, views.py 변경, models/init 파일 정리

Kang9805 committed:
🐛 fix : applicant_application_view 도메인 전체 500 오류 원인 수정

recruitment_uuid를 Recruitment FK(id)가 아닌 uuid로 비교하던 오류 수정

Meoyoug commented on apps/application/views/applicant_application_views.py:
```diff
@@ -0,0 +1,117 @@
+        responses={200: dict},
+        tags=["Application - Applicant"],
+    )
+    def post(self, request: Request, recruitment_uuid: str) -> Response:
```
recruitment_uuid가 존재하지 않는 경우에 대한 예외처리가 없네요

Meoyoug commented on apps/application/views/applicant_application_views.py:
```diff
@@ -0,0 +1,117 @@
+        responses={200: ApplicantApplicationListSerializer},
+        tags=["Application - Applicant"],
+    )
+    def get(self, request: Request) -> Response:
```
<img width="1339" height="454" alt="Image" src="https://github.com/user-attachments/assets/34c143e5-59b6-40fb-b61d-cd41c1873840" />

이게 성연님이 작성하신 응답 스키마이고

api 명세에서 요구하는 응답 스키마는

<img width="569" height="439" alt="Image" src="https://github.com/user-attachments/assets/39fa085a-651c-4cb9-9579-2608457a1ead" />

이렇게 인데 혹시 api분리를 고려하고 계시는건지 누락된건지 확인해보시고 말씀부탁드려요

Meoyoug commented on apps/application/views/applicant_application_views.py:
```diff
@@ -0,0 +1,117 @@
+        application.save(update_fields=["status"])
+
+        serializer = ApplicationCancelSerializer(application)
+        return Response(serializer.data, status=status.HTTP_200_OK)
```
취소된 지원내역의 uuid, 지원내역 상태를 응답으로 반환하는것 또한 api 명세랑은 다른 것 같은데 의도하신 건가요?

Kang9805 committed:
♻️ refactor : application serializer, admin serializer 수정, cancel serializer 제거

application serializer 수정
admin application serializer 수정
cancel application serializer 제거

Kang9805 committed:
♻️ refactor : applicant_application_view 수정

recruitment_uuid 존재하지 않을 시 예외처리
api 명세서에 맞는 응답 스키마 적용
취소 지원내역 반환 응답 수정

거

Kang9805 committed:
♻️ refactor : applicant_urls 수정

urls 매핑 뒤에 name으로 별칭 부여

Kang9805 committed:
✅ test : test_applicant_application_views 작성

test_applicant_application_views 작성
